### ISSUE TITLE
DATA: Refugee Resource/Property Model

### ISSUE THREAD

**Issue by Spectralgo** [hassanhabib/Refugee.Core#23]

**Comment by bramburn** [comment]
question on this, can a refugee's family member not be in the same country as the refugee family? I.e. split group? so mum outside the country, dad in the conflict zone

**Comment by bilalmehrban** [comment]
Yes, it can be. What can be done is it that we can have refugee family group, which have family members that may be across different countries.

**Comment by Spectralgo** [comment]
We have a current location property on the refugee class. And instead of creating a family class we made a RefugeeGroup. 

We might add a class for a proper family. Like that the family can be split into refugee groups or we can even add non refugee family members to the family.

We'll definitely drill down this business case once we're done building a MVP.

Being:
Host offers shelter to RefugeeGroup (e.g. family, orphans)

Thx for those comments. It helps me think in a macroscopic way.

**Comment by bilalmehrban** [comment]
@Spectralgo do we have ERD for this? if not why dont we make one using draw.io, and share with group. That would be super helpfull.

**Comment by Spectralgo** [comment]
Thanks for asking. I thought about it and I came to this conclusion.

Hassan gave us a brief story to implement during our first design/architecture session. It was a quick draft on draw.io, but it was sufficient to start working upon.

The first goal is to have implemented the following story:

Host has Shelter.

RefugeeGroup has list of refugees.

Host makes **shelter offer** 

RefugeeGroup **request shelter** (based on list of shelter offers).

Host accepts **shelter request**.

Hassan has a lot of experience on system design and it would be great content for his channel.

Having our hands in the code right now, makes us think, almost unconsciously, about potential features and solutions. We are already drawing drafts in our minds.

Knowing all of this, it seems right to finish implementing this story. We'll have some ground on which Hassan will be able to reflect on, with us. We would be able during this session to argue with him about architectural ideas. At this point, we should be able to produce good diagrams and share them in our docs. Our train of thought would be recorded and we would have gain enough unconscious knowledge to fight back against bad ideas.

So, this is definitely something we need for everyone to see, but not quite yet. 

Do you agree ?

Thanks again, it forces me to think properly 🤔

**Comment by bilalmehrban** [comment]
That quick draft was about the overall design of the system. Hassan do have the experiance in System design that is ofcourse a final go on the design/architecture of this system. 

But my point is that having an ERD of the entites is pretty much helpful. Any one who is going to join and contribute to the project can have a look on the ERD and know about the Entites and there relationships. 

Right now the MVP is not going to be very complex or huge so it would be easier to make an ERD and than going forward update it as changes are made. 

It will also be helpfull in cases where some one else is going to work on an entity who was not present in the session. 

Even if currently you don't want to share it with all, it will still be helpful for the colabrators to disucss and update the entity structure.  Potential drafts in mind can easily be lost for example if we get into some other work.

I hope that you got my point on this. 

Thanks 😊

**Comment by Spectralgo** [comment]
Yeah you did, and I'll provide a link to our GitBook. The GitBook should eventually linked to this GitHub repo or another. 

I will make a section to allow drafts of uml to be consulted. But please, be careful, don't put to much work into it, not right now. I just don't want you to invest a lot too early, and be disappointed because we basically never built what you had design. 

Kaps loves UMLs too so you should try to get in touch together. It would be cool to see you work together on those.

**Comment by Spectralgo** [comment]
@bilalmehrban did you add me on discord?

**Comment by bilalmehrban** [comment]
@Spectralgo It would be collective effort. It would be hardly 15/30 mins per sprint. as it not a complex system design. 

No, I have not added yet. Will add you asap.

**Comment by Spectralgo** [comment]
@rcruzfreelance we were talking about the diagrams today
